orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/ExecutionRunnerSupport.java

```diff
@@ -32,30 +36,39 @@ public ExecutionRunnerSupport(Collection<StageDefinitionBuilder> stageDefinition
     this.stageDefinitionBuilders = stageDefinitionBuilders;
   }
 
-  @Override public <T extends Execution> void start(T execution) throws Exception {
+  @Override
+  public <T extends Execution> void start(T execution) throws Exception {
     List<Stage<T>> stages = new ArrayList<>(execution.getStages()); // need to clone because we'll be modifying the list
     stages.stream().forEach(this::planStage);
   }
 
   protected <T extends Execution> void planStage(Stage<T> stage) {
     StageDefinitionBuilder builder = findBuilderForStage(stage);
-    builder
-      .preStages(stage)
-      .forEach(preStage -> {
-        List<Stage> stages = stage.getExecution().getStages();
-        int index = stages.indexOf(stage);
-        stages.add(index, preStage);
-        planStage(preStage);
-      });
+    Map<SyntheticStageOwner, List<Stage<T>>> allStages = builder
+      .aroundStages(stage)
+      .stream()
+      .collect(Collectors.groupingBy(Stage::getSyntheticStageOwner));
+
     reverse(
-      builder
-        .postStages(stage))
-      .forEach(postStage -> {
-        List<Stage> stages = stage.getExecution().getStages();
-        int index = stages.indexOf(stage);
-        stages.add(index + 1, postStage);
-        planStage(postStage);
-      });
+      allStages.getOrDefault(STAGE_BEFORE, Collections.emptyList())
```

Comment:
There is some historical preference that the order of STAGE_BEFORE's should be reversed.

Aside from digging through code, I'm not sure it's clear to anyone implementing a stage that this is the way it works. I've filed https://jira.netflix.com/browse/SPIN-1768 to circle around on this ... 

https://jira.netflix.com/browse/SPIN-1768